converter.go

```diff
@@ -385,6 +385,21 @@ func arrayToString(v driver.Value, tsmode snowflakeType, params map[string]*stri
 		return bindingValue{&res, "json", &schemaForBytes}, nil
 	} else if isSliceOfSlices(v) {
 		return bindingValue{}, errors.New("array of arrays is not supported")
+	} else if valuer, ok := v1.Interface().(driver.Valuer); ok { // check for driver.Valuer satisfaction and honor that first
+		value, err := valuer.Value()
+
+		if err != nil || value == nil {
+			return bindingValue{}, err
+		}
+
+		if v, ok := value.(string); ok {
+			return bindingValue{&v, "", nil}, nil
+		}
+
+		return bindingValue{}, nil
+	} else if stringer, ok := v1.Interface().(fmt.Stringer); ok { // alternate approach; check for stringer method. Guarantees it's String() and returns string
+		value := stringer.String()
+		return bindingValue{&value, "", nil}, nil
 	}
 	res, err := json.Marshal(v)
```

Comment:
After this, the bindingValue is set to "json". Should that be the case, or should it be string? Since it is already Marshaled, why set it to json again? That will cause it to be marshaled a second time. It feels like it should not marshal here if it also sets the binding value to json.

Reply:
As mentioned above - this is only a hint for backend, how to treat this string value - either as a string literally or as a structured type described in json.